tsdb/cursor.go

```diff
@@ -297,13 +303,27 @@ type pointHeapItem struct {
 }
 
 type pointHeap []*pointHeapItem
+type pointHeapReverse struct {
+	pointHeap
+}
 
-func newPointHeap() *pointHeap {
+func newPointHeap() heap.Interface {
```

Comment:
Only thing you might prefer is that the "constructor" could be of the form `newPointHeap(ascending bool)`.